### DISCUSSION TITLE
New Source: Rippling Connector

### DISCUSSION THREAD

**Discussion by josh-orellana** [airbytehq/airbyte#35260] [New Connector Request]
## Tell us about the new connector you’d like to have
* Which source and which destination? **Rippling Source for Snowflake** 
* Do you need a specific version of the underlying data source e.g: you specifically need support for an older version of the API or DB? **Nope**

## Describe the context around this new connector
* Which team in your company wants this integration, what for? This helps us understand the use case. HR team would like to calculate D&I OKRs and Finance team would like to calculate payroll by department
* How often do you want to run syncs? 24 hours
* If this is an API source connector, which entities/endpoints do you need supported?  Companies/Employees/Payroll

## Describe the alternative you are considering or using
What are you considering doing if you don’t have this integration through Airbyte? Manually shooting a CSV into the BI tool

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202102171868563/1202126677454407) by [Unito](https://www.unito.io)

**Comment by josh-orellana** [comment]
There's also a Postman collection
https://documenter.getpostman.com/view/11475460/T1LQhS1V?version=latest

**Comment by marcosmarxm** [comment]
Awesome @jorellana92 do you want to try and contribute creating this connector? In any case I'll add it to the connector roadmap.

**Comment by cheyura** [comment]
### Integration Vetting

**Webhook-based? (no/partially/yes)**
Partillay https://developer.rippling.com/docs/rippling-api/docs/Getting-Started/h-Webhooks.md


Available authentication modes (API key/Oauth/other)
Rippling integrations rely on OAuth 2.0, in which Rippling is the server and your application is the client. At a high-level Rippling will send the user to your redirect URL with a ?code=<xxx> parameter when they install your application on Rippling. This provides the authorization that your application will utilize to exchange for access and refresh tokens.
https://developer.rippling.com/docs/rippling-api/docs/Getting-Started/f-Authorization.md


Creating an account
Needs communication with support,prod starts from 8usd per month


Has Sandbox environment ?
Needs clarification


How to populate the account with data?
UI,api

Available streams for sync

A. Companies

B. Employees

C. Payroll

D. 401k

F. Groups

G. SAML

E. Current User

H. ATS

I. Application Management

J. Reports

Integration supports incremental sync?

The most reliable method to ingest all activity from Rippling is to use a pagination cursor via the next parameter. This will ensure that events are not skipped or duplicated due to the lack of timestamp precision.


Other information/blockers
Meeting with support is needed

**Comment by cheyura** [comment]
@sherifnada they cannot offer sandbox/ please see inbox mail

**Comment by sherifnada** [comment]
Rippling doesn't offer sandbox accounts. Since we can't replicate a test environment for this data, we're putting it on hold.

**Comment by anaselmhamdi** [comment]
Hey team,
Anas from Gorgias here, we've already developed a connector for Rippling, I'd be glad to contribute !

**Comment by anaselmhamdi** [comment]
[PR opened here](https://github.com/airbytehq/airbyte/pull/27242) team, let me know if you need anything else !

[a comment by an automated account is not shown]
